client/ng/components/accounts/register/register.component.js

```diff
@@ -0,0 +1,61 @@
+// jshint esversion: 6
+
+import register from './register.module';
+import template from './register.html';
+import {
+    account
+} from '../accounts.js';
+
+class RegisterForm {
+    constructor($state) {
+        'ngInject';
+        this.user = {};
+
+        this.goToLogin = () => {
+         $state.go("login");
+       };
+    }
+
+    submit(valid) {
+      var message;
+      console.log(this.user.terms);
+
+      if (valid){
+        if (this.user.password === this.user.password2) {
+            var Object = {
```

Comment:
Właśnie nadpisałeś globalną funkcję (konstruktor) Object. Nie wiem jaki wpływ to ma na resztę aplikacji, ale definitywnie  nazwa zmiennej do zmiany. Poza tym zwykła zmienna nie powinna być pisana z wielkiej litery.